aerogear/graphback
# Typesafe resolvers

Add ability to pass graphql-code-gen typings into resolvers to make them typesafe

Api looks like this
```
  const backend: IGraphQLBackend = {};
    // FIXME - this is for testing only
    // tslint:disable-next-line: no-parameter-reassignment
    resolverOptions = {
      format: 'ts',
      types: {
        typesImportStatement: 'import { Resolvers } from "../../generated-types"',
        resolverType: 'Resolvers',
      }
    };
    const schemaGenerator = new SchemaGenerator(this.inputContext, tsSchemaFormatter)
    backend.schema = schemaGenerator.generate()

    backend.resolvers = generateResolvers(this.inputContext, resolverOptions);
```

## Verification

I have added temporary step (as our cli do not allow to use codegen).

1. Execute graphback generate (from debug or cli)
2. Execute `yarn graphql codegen` for the typings
3. Code should compile -> `yarn start`


I have also added js format for supporting js templates
